pkg/asset/installconfig/ovirt/credentials.go

```diff
@@ -165,104 +196,48 @@ func engineSetup() (Config, error) {
+	err = HTTPResource.checkURLResponse()
 	if err != nil {
 		return EngineConfig, err
 	}
 
-	// Check if CA is trusted locally
-	var importCACert bool = false
-
-	// Store if connection with Engine is secure
-	var ConnectionSecure bool = false
+	// Set Engine PEM URL for Download
+	EngineConfig.PemURL = fmt.Sprintf(
+		"https://%s/ovirt-engine/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA",
+		EngineConfig.FQDN)
+	logrus.Debug("PEM URL: ", EngineConfig.PemURL)
 
-	ConnectionSecure, err = checkCATrust(EngineConfig.FQDN)
+	// Create tmpFile to store the Engine PEM file
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "engine-")
 	if err != nil {
-		return EngineConfig, err
-	}
-
-	if ConnectionSecure {
-		EngineConfig.Insecure = false
-	} else {
-		EngineConfig.Insecure = true
-		message := fmt.Sprintf("Would you like to import Engine CA cert locally from: %s ?", EngineConfig.PemURL)
-		err = survey.AskOne(
-			&survey.Confirm{
-				Message: message,
-				Default: false,
-				Help:    "In order to securly communicate with the oVirt engine, the certificate authority must be trusted by the local system.",
-			},
-			&importCACert,
-			nil)
-		if err != nil {
-			return EngineConfig, err
-		}
-	}
-
-	// If users request, let's work to import Engine CA locally
-	if importCACert == true {
-		logrus.Info("Downloading Engine CA cert from Engine...")
-		tmpFile, err := ioutil.TempFile(os.TempDir(), "engine-")
-		if err != nil {
-			fmt.Println("Cannot create temporary file", err)
-		}
-		defer os.Remove(tmpFile.Name())
-
-		logrus.Debugf("CA cert temporary stored: %s", tmpFile.Name())
-
-		/* curl command */
-		_, err = execCommand(CurlCommand, "-k", EngineConfig.PemURL, "-o", tmpFile.Name())
-		if err != nil {
-			return EngineConfig, err
-		}
-		logrus.Debugf("PEM file: %s", tmpFile.Name())
-
-		var content []byte
-		content, err = fileContent(tmpFile.Name())
-		if err != nil {
-			return EngineConfig, err
-		}
-		logrus.Debug(string(content))
-
-		/* Check if CA file already exists */
-		CaPath := fmt.Sprintf("/etc/pki/ca-trust/source/anchors/%s.pem", strings.ReplaceAll(EngineConfig.FQDN, ".", "-"))
-		if _, err := os.Stat(CaPath); err == nil {
-			return EngineConfig, err
-		}
-
-		/* Copy the CA to anchors */
-		_, err = execCommand(SudoCommand, CpCommand, tmpFile.Name(), CaPath)
-		if err != nil {
-			return EngineConfig, err
-		}
-
-		/* chmod to allow non root users to read the cert */
-		_, err = execCommand(SudoCommand, ChmodCommand, "0644", CaPath)
-		if err != nil {
-			return EngineConfig, err
-		}
-
-		/* Update CA database */
-		_, err = execCommand(SudoCommand, "/usr/bin/update-ca-trust")
-		if err != nil {
-			return EngineConfig, err
+		fmt.Println("Cannot create temporary file", err)
+	}
+	defer os.Remove(tmpFile.Name())
+
+	// Download PEM
+	HTTPResource.saveFilePath = tmpFile.Name()
+	HTTPResource.skipVerify = true
+	HTTPResource.urlAddr = EngineConfig.PemURL
+	err = HTTPResource.downloadFile()
+	if err == nil {
```

Comment:
Why not use the regular:
err != nil ==> print something or return an error
syntax, I see that if err is not nil we are not doing anything about it

Reply:
agree